### ISSUE TITLE
Create a Flatpak package and publish it on Flathub

### ISSUE THREAD

**Issue by Newbytee** [GTcreyon/SM63Redux#60]
**Describe the feature you'd like**
I would like to see SM63Redux published as a Flatpak on [Flathub](https://flathub.org/home). This would make it easy to install on Linux desktops and other devices such as the Steam Deck.

**Do you intend to contribute this yourself? If so, how do you plan on doing it?**
I can help out with it, but ideally I would like it if you could do the final Flathub submission as Flathub prefers if the authors of the program publish it there.

**Additional context**
Information about app submission  on Flathub's GitHub Wiki: https://github.com/flathub/flathub/wiki/App-Submission

**Comment by GTcreyon** [comment]
Not a bad idea, but I'm a little wary of the legal implications. I'm not sure how Flathub would feel about fangames.

Also, I don't know how useful it would be, given that Redux is just an executable file that you can download, not something that has to be installed.
If you'd like a quick easy way to download the current version from the terminal, try this command:
```
wget 'https://drive.google.com/uc?export=download&id=14AoZDbMLjkjab3Rb_0OXv4d2jMfaFYis&export=download&confirm=t' -O temp.zip && unzip temp.zip && rm temp.zip```

That aside, I'm not against it, but I'd need some convincing.

**Comment by Newbytee** [comment]
> Not a bad idea, but I'm a little wary of the legal implications. I'm not sure how Flathub would feel about fangames.

For what it's worth, there's various Sonic fangames on Flathub right now: https://flathub.org/apps/search/sonic

**Comment by GTcreyon** [comment]
> For what it's worth, there's various Sonic fangames on Flathub right now: https://flathub.org/apps/search/sonic

Indeed, but no Mario fangames. So I'm a little wary, based on Nintendo's policies.
That said, it's good to know that Flathub isn't directly opposed to it.
Also, an AUR package isn't out of the question. Excluding pre-built binaries feels safer.

**Comment by jaschutte** [comment]
Not related to flathub, but I feel still relevant to the larger subject is a possible inclusion in the F-Droid store. An FOSS only Appstore for android.

**Comment by Newbytee** [comment]
>  If you'd like a quick easy way to download the current version from the terminal, try this command:

It's not so much about me really as it is about the wider Linux ecosystem. I believe we have something very good going with Flatpak and Flathub, and getting great games like SM63Redux on there is a good way to support that.

> That aside, I'm not against it, but I'd need some convincing.

Personally, I like that Flatpak packages give me control over what parts of the systems programs have access to. I also like how it means updates are handled by one central program instead of every program handling them by itself. Additionally, Flatpaks work even on unconventional distributions like Alpine Linux (as it uses musl).

**Comment by GTcreyon** [comment]
> I believe we have something very good going with Flatpak and Flathub

Who is "we"?

> [...] and getting great games like SM63Redux on there is a good way to support that.

I don't mean to be rude, but I don't think we as a team should be making decisions for Redux based on how they would help *other* projects. I'm all for an altruistic approach to development, but I don't think benefiting a specific distribution system by contributing to their software base is altruistic, so I think this is sort of a non-argument. If we're going to do it, it should be based on utility to our users, not to Flathub.

> Personally, I like that Flatpak packages give me control over what parts of the systems programs have access to.

That's cool, but are access permissions really relevant to Redux? All we do is send a few network requests and (in future) store some config files.

> I also like how it means updates are handled by one central program instead of every program handling them by itself.

This is indeed a strong positive, but not so relevant right now, since we're not really updating the game frequently. We might revisit this later

> Additionally, Flatpaks work even on unconventional distributions like Alpine Linux (as it uses musl).

I don't know much about that, but I'd advise just using gcompat or whatever for that. I don't think enough of our users run Alpine or similar to make this worth the effort. The Linux build is already pretty unpopular.

**Comment by GTcreyon** [comment]
> Not related to flathub, but I feel still relevant to the larger subject is a possible inclusion in the F-Droid store. An FOSS only Appstore for android.

I would make a separate issue for this, but I'm similarly indifferent on this topic. It might aid convenience slightly.

**Comment by Newbytee** [comment]
> Who is "we"?

As in, the broader Linux community.

> I don't mean to be rude, but I don't think we as a team should be making decisions for Redux based on how they would help other projects. I'm all for an altruistic approach to development, but I don't think benefiting a specific distribution system by contributing to their software base is altruistic, so I think this is sort of a non-argument. If we're going to do it, it should be based on utility to our users, not to Flathub.

That was intended to explain why I submitted this issue rather than just using the download, not why you should put this game on Flathub. I agree that me liking Flathub is not a good reason for you to put your game on Flathub.

> That's cool, but are access permissions really relevant to Redux? All we do is send a few network requests and (in future) store some config files.

As I see it, it is. Flatpak lets you transparently show users that this indeed only is what your game does as you only declare the permissions you need in the manifest. When I knew less about computers I was very hesitant about downloading fangames chiefly because I didn't know what they would do to my computer.

**Comment by GTcreyon** [comment]
> That was intended to explain why I filed this issue rather than just using the download, not why you should put this game on Flathub. I agree that me liking Flatpak/Flathub is not a good reason for you to put your game on Flathub.

Right, right, I understand now.

> As I see it, it is. Flatpak lets you transparently show users that this indeed only is what your game does as you only declare the permissions you need in the manifest.

That's true. I can see the benefit in that.

**Comment by Newbytee** [comment]
Another option could be to just release a standalone .flatpak file. It would have the desktop integration benefits of Flatpak while letting you handle the distribution of it yourself.

**Comment by GTcreyon** [comment]
> Another option could be to just release a standalone .flatpak file. It would have the desktop integration benefits of Flatpak while letting you handle the distribution of it yourself.

A good idea. Could use that as a bridge for testing the waters.
